Tools/YamlCreate.ps1

```diff
@@ -1,262 +1,261 @@
 }
 catch {
-    write-host "Error downloading file. Please run the script again." -ForeGroundColor red
+    Write-Host "Error downloading file. Please run the script again." -ForegroundColor Red
     exit 1
 }
+finally {
+    $Stream.Close()
+}
 
+Write-Host "Url: $URL"
+Write-Host "Sha256: $Hash"
 
-# This command will get the sha256 hash
-$Hash = get-filehash -InputStream $stream
-$stream.Close()
-
-
-$string = "Url: " + $URL  ;
-Write-Output $string
-$string = "Sha256: " + $Hash.Hash
-$string
-$OFS
-write-host "File downloaded. Please Fill out required fields. "
+Write-Host $OFS
+Write-Host "File downloaded. Please Fill out required fields."
 
+#endregion
 ##########################################
-# Read in metadata
+
 ##########################################
+#region Read in metadata
 
-While ($id.Length -lt 4 -or $id.length -ge 255) {
-    write-host  'Enter the package Id, in the following format <Publisher.Appname>'
-    $id = Read-Host -Prompt 'For example: Microsoft.Excel'
+while ($ID.Length -lt 4 -or $ID.Length -ge 255) {
+    Write-Host 'Enter the package Id, in the following format <Publisher.Appname>'
+    $ID = Read-Host -Prompt 'For example: Microsoft.Excel' | TrimString
 }
 
 $host.UI.RawUI.ForegroundColor = "White"
-While ($publisher.Length -eq 0 -or $publisher.length -ge 128) {
-    $publisher = Read-Host -Prompt 'Enter the publisher'
+while ([string]::IsNullOrWhiteSpace($Publisher) -or $Publisher.Length -ge 128) {
+    $Publisher = Read-Host -Prompt 'Enter the publisher' | TrimString
 }
 
-While ($AppName.Length -eq 0 -or $AppName.length -ge 128) {
-    $AppName = Read-Host -Prompt 'Enter the application name'
+while ([string]::IsNullOrWhiteSpace($AppName) -or $AppName.Length -ge 128) {
+    $AppName = Read-Host -Prompt 'Enter the application name' | TrimString
 }
 
-While ($version.Length -eq 0) {
-    $version = Read-Host -Prompt 'Enter the version. For example: 1.0, 1.0.0.0'
-    $filename = $version + ".yaml"
+while ([string]::IsNullOrWhiteSpace($version)) {
+    $version = Read-Host -Prompt 'Enter the version. For example: 1.0.0, 1.0.0.0, 1.0' | TrimString
+    $ManifestName = $version + ".yaml"
 }
 
-While ($License.Length -eq 0 -or $License.length -ge 40) {
-    $License = Read-Host -Prompt 'Enter the License, For example: MIT, or Copyright (c) Microsoft Corporation'
+while ([string]::IsNullOrWhiteSpace($License) -or $License.Length -ge 40) {
+    $License = Read-Host -Prompt 'Enter the License, For example: MIT, or Copyright (c) Microsoft Corporation' | TrimString
 }
 
-While ($InstallerType -notin ("exe", "msi", "msix", "inno", "nullsoft", "appx", "wix", "zip")) {
-    $InstallerType = Read-Host -Prompt   'Enter the InstallerType. For example: exe, msi, msix, inno, nullsoft'
+while ($InstallerType -notin @("exe", "msi", "msix", "inno", "nullsoft", "appx", "wix", "zip")) {
+    $InstallerType = Read-Host -Prompt 'Enter the InstallerType. For example: exe, msi, msix, inno, nullsoft'
```

Comment:
Is there a reason this line doesn't have | TrimString?

Reply:
Yes, because here the value must be equal to one of the array elements. Think of it like an enum where only expected values can be specified.